src/common/components/ListAllContacts.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import {
+  Image,
+  StyleSheet,
+  Text,
+  View,
+  FlatList,
+  TouchableOpacity,
+} from 'react-native';
+
+import {useNavigation} from '@react-navigation/native';
+
+import {allContacts} from '../../api/fakeApiUser';
+import routeConstants from '../../utils/constants/routeConstants';
+import typography from '../../themes/typpography';
+import {colors} from '../../themes/theme';
+import {scale} from 'react-native-size-matters';
+
+const ListAllContacts = () => {
+  const navigation = useNavigation();
+
+  const handlePress = () => {
+    navigation.navigate(routeConstants.DETAILS);
+  };
+
+  const renderContact = ({item}) => (
+    <ContactItem contact={item} handlePress={handlePress} />
+  );
+  return (
+    <>
+      <View style={styles.mainView}>
+        <FlatList
+          data={allContacts}
+          renderItem={renderContact}
+          keyExtractor={item => item.id}
+        />
+      </View>
+    </>
+  );
+};
+
+const ContactItem = ({contact, handlePress}) => (
+  <TouchableOpacity style={styles.userContainer} onPress={handlePress}>
+    <Image
+      style={styles.profileImage}
+      source={require('../../assets/images/dummy_image.png')}
+    />
+    <View style={styles.userDataContainer}>
+      <Text style={typography.h3}>{contact.name}</Text>
+      <Text style={typography.body}>{contact.email}</Text>
+      <Text style={typography.body}>{contact.phone}</Text>
+    </View>
+    <Image
+      style={styles.right_arrow}
+      source={require('../../assets/images/right_arrow.png')}
+    />
+  </TouchableOpacity>
+);
```

Comment:
We could have moved this `ContactItem` into a separate common component so in future we can use it at different place. 
`"Try think for long term"`